vsintegration/src/FSharp.Editor/LanguageService/Tokenizer.fs

```diff
@@ -736,15 +740,14 @@ module internal Tokenizer =
 
     let private getCachedSourceLineData(documentKey: DocumentId, sourceText: SourceText, position: int, fileName: string, defines: string list) = 
         let textLine = sourceText.Lines.GetLineFromPosition(position)
-        let textLinePos = sourceText.Lines.GetLinePosition(position)
-        let lineNumber = textLinePos.Line + 1 // FCS line number
+        let textLinePos = sourceText.Lines.GetLinePosition(position)      
         let sourceTokenizer = FSharpSourceTokenizer(defines, Some fileName)
         let lines = sourceText.Lines
         // We keep incremental data per-document. When text changes we correlate text line-by-line (by hash codes of lines)
         let sourceTextData = getSourceTextData(documentKey, defines, lines.Count)
         // Go backwards to find the last cached scanned line that is valid
         let scanStartLine = 
-            let mutable i = min (lines.Count - 1) lineNumber
+            let mutable i = min (lines.Count - 1) textLinePos.Line
```

Comment:
So this was a bug? Or why it's now not `textLinePos.Line + 1`? Just curious.

Reply:
It was bug, line data is 0-indexed in objects coming from Roslyn, as well as it is in our caches within the Tokenizer.